### ISSUE TITLE
Get-ChildItem fails on Junction/Symlink when it points to an UNC directory

### ISSUE THREAD

**Issue by ThatMartinScott** [PowerShell/PowerShell#19368]
### Prerequisites

- [X] Write a descriptive title.
- [X] Make sure you are able to repro it on the [latest released version](https://github.com/PowerShell/PowerShell/releases)
- [X] Search the existing issues.
- [X] Refer to the [FAQ](https://github.com/PowerShell/PowerShell/blob/master/docs/FAQ.md).
- [X] Refer to [Differences between Windows PowerShell 5.1 and PowerShell](https://docs.microsoft.com/powershell/scripting/whats-new/differences-from-windows-powershell).

### Steps to reproduce

Create Symlink or Junction using an UNC _(i.e. \\my-pc\volume1)_ as the target. Using any tool: (junction.exe, mklink.exe, New-Item) .  
Example:
```
C:\> Junction -s TestLink  \\my-pc\volume1\users\me\TestFolder
C:\> cd TestLink
C:\TestLink> dir                                   <-fails - Incorrect Parameter Error
_Get-ChildItem: The parameter is incorrect. : 'C:\Users\ThatM\Offline'                     <--Error_
C:\TestLink> Get-ChildItem *.*                          <-fails

```
Any Get-ChildItem call to the Junction/SymLink will fail with same error, example:
`Get-Children C:\TestLink\                    <-fails`

Even though Get-ChildItem cannot list the Items in the Junction/SymLink, it can set location to any child folder. Example:
```
C:\> cd TestLink
C:\TestLink> cd TestSub
C:\TestLink\TestSub> Get-ChildItem *.*                     -works in child folder of Link, but not in C:\TestLink
```

I tried a bunch of other Powershell commands to see if they would fail, and it is only Get-ChildItem that is broken.
i.e. Get-Item works inside the C:\TestLink folder.

Additional Info: 
Get-ChildItem with targets that use drive letters (C:\LocalLink -> C:\Users) work as expected. It is only a UNC target that causes Get-ChildItem to fail and it doesn't matter if it points to the local system or another PC.  `(C:\TestLink -> \\my-pc\volume1\TestFolder) [fail]    (C:\TestLink -> \\server-pc\volume1\TestFolder) [fail]`

This is a fairly new bug, as this used to work just fine a month or more ago.  Get-ChildItem in Windows Powershell works just fine with Junction/SymLinks to an UNC directory.

To be complete and clear, it is only 7.x of Powershell, and it is only Get-ChildItem.  
(Could be any 7.x point release - I wish I had noticed earlier.)

Everything else works with these Junction/Symlinks, i.e.: CMD, Explorer, C# Code, C++ NTFS Calls.

**This is kind of a big break/issue, as Symbolic Links that point to an UNC are used a lot in Enterprise and IT realms - things that go back to 1999, like Profile locations, Shell Folders, etc.**   

### Expected behavior

```console
PS C:\>  ni -type SymbolicLink MyLinkFolder \\my-pc\volume2\MyStuff
PS C:\> cd MyLinkFolder
PS C:\MyLinkFolder> dir *.*

    Directory: C:\MyLinkFolder

Mode                 LastWriteTime         Length Name
----                 -------------         ------ ----
d----            3/7/2023  3:10 AM                My Sub Folder
d----           3/18/2023 10:04 PM                coding101.txt
d----           3/18/2023  9:31 PM                CoolApp.exe

PS C:\MyLinkFolder>
```

### Actual behavior

```console
PS C:\> ni -type SymbolicLink MyLinkFolder \\my-pc\volume2\MyStuff
PS C:\> cd MyLinkFolder
PS C:\MyLinkFolder> dir *.*
Get-ChildItem: The parameter is incorrect. : 'C:\MyLinkFolder'
PS C:\MyLinkFolder>
```

### Error details

```console
Get-ChildItem: The parameter is incorrect. : 'C:\MyLinkFolder'
```

### Environment data

```powershell
PS C:\Users\Me> $PSVersionTable

Name                           Value
----                           -----
PSVersion                      7.3.3
PSEdition                      Core
GitCommitId                    7.3.3
OS                             Microsoft Windows 10.0.22624
Platform                       Win32NT
PSCompatibleVersions           {1.0, 2.0, 3.0, 4.0…}
PSRemotingProtocolVersion      2.3
SerializationVersion           1.1.0.1
WSManStackVersion              3.0
```

### Visuals

![Dir of Symbolic Links](https://user-images.githubusercontent.com/109339755/226222285-787d25fe-575c-46b3-aaf8-d59879fa4c11.png)
![Dir Get-ChildItem Failing](https://user-images.githubusercontent.com/109339755/226222299-a80c489f-6164-45f3-a5b9-6d003e00d4ef.png)

**Comment by SteveL-MSFT** [comment]
This seems to be working in the daily build.  Looking at the error details:

```powershell
Exception             :
    Type       : System.IO.IOException
    TargetSite :
        Name          : CreateDirectoryHandle
        DeclaringType : System.IO.Enumeration.FileSystemEnumerator`1[TResult]
        MemberType    : Method
        Module        : System.Private.CoreLib.dll
    Message    : The parameter is incorrect. : 'C:\Users\slee\test\testlink'
    Source     : System.Private.CoreLib
    HResult    : -2147024809
    StackTrace :
   at System.IO.Enumeration.FileSystemEnumerator`1.CreateDirectoryHandle(String path, Boolean ignoreNotFound)
   at System.IO.Enumeration.FileSystemEnumerator`1.Init()
   at System.IO.Enumeration.FileSystemEnumerable`1..ctor(String directory, FindTransform transform, EnumerationOptions options, Boolean isNormalized)
   at System.IO.Enumeration.FileSystemEnumerableFactory.DirectoryInfos(String directory, String expression, EnumerationOptions options, Boolean isNormalized)
   at System.IO.DirectoryInfo.InternalEnumerateInfos(String path, String searchPattern, SearchTarget searchTarget, EnumerationOptions options)
   at System.IO.DirectoryInfo.EnumerateDirectories()
   at Microsoft.PowerShell.Commands.FileSystemProvider.Dir(DirectoryInfo directory, Boolean recurse, UInt32 depth, Boolean nameOnly, ReturnContainers returnContainers, InodeTracker tracker)
TargetObject          : C:\Users\slee\test\testlink
CategoryInfo          : ReadError: (C:\Users\slee\test\testlink:String) [Get-ChildItem], IOException
FullyQualifiedErrorId : DirIOError,Microsoft.PowerShell.Commands.GetChildItemCommand
InvocationInfo        :
    MyCommand        : Get-ChildItem
    ScriptLineNumber : 1
    OffsetInLine     : 1
    HistoryId        : 4
    Line             : dir testlink
    Statement        : dir testlink
    PositionMessage  : At line:1 char:1
                       + dir testlink
                       + ~~~~~~~~~~~~
    InvocationName   : dir
    CommandOrigin    : Internal
ScriptStackTrace      : at <ScriptBlock>, <No file>: line 1
PipelineIterationInfo :
```

The problem was in an underlying .NET API, and it appears to have been fixed and would show up in next 7.4 preview.

**Comment by ThatMartinScott** [comment]
Confirming that it is fixed.   
I am such a goof; when I didn't find a reference to the bug, I never thought to check the current builds. 
-Thank you for taking time to check it out.
